conans/client/manager.py

```diff
@@ -461,6 +462,18 @@ def build(self, conanfile_path, source_folder, build_folder, package_folder, tes
                                  "Create a '%s' and move manually the "
                                  "requirements and generators from '%s' file"
                                  % (CONANFILE, CONANFILE, CONANFILE_TXT))
+
+        if test:
+            conan_info_path = os.path.join(build_folder, CONANINFO)
```

Comment:
If I understood well the missing require is to the package being tested, right? why don't just:
```
if test:
    conan_file.requires.add(str(test))
```

Reply:
It might already have been defined. The test_package/conanfile might declare also other requirements. Might try to just add it and fail try-except if the requirement is already duplicated. Will check it.

Reply:
Yes, but you can check it after adding it. What I don't understand is why to parse the `conaninfo` if the only missing require should be the recipe being tested.

Reply:
Yes, good catch, much better, I have submitted changes.